chore (database): configure PostgreSQL as DB Engine

lennykioko opened the pull request:
### **What does this PR do?**
Configure PostgreSQL as DB Engine

### **Description of the task to be completed**
- Configure settings for integration with PostgreSQL
- Add pyscopg2

### **How should this be manually tested**
The application integrates smoothly with PostgreSQL once migrations have been run

### **Any background context that you want to provide**
PostgreSQL is Open Source, fully featured and optimized for production-ready environments as compared to SQLite that comes as default with the Authors Haven application

### **What are the relevant pivotal tracker stories**
#159053979

Commit:
chore (database): configure PostgreSQL as DB Engine

- modify database settings
- add pyscopg2

[Delivers #159053979]

dmigwi commented on authors/settings.py:
```diff
@@ -81,8 +81,12 @@
+        'USER': 'team-scorpion',
+        'PASSWORD': 'password',
+        'HOST': 'localhost',
+        'PORT': '',
```
I would like you to implement this with an abstraction that hides sensitive information like `NAME`,  `USER`, `PASSWORD`,  `LOCALHOST` and `POST` in github. Exposing that information is a security lapse.
Check this implementation here on how to add a .env file with such confidential information. https://gist.github.com/bennylope/2999704

dmigwi commented on .gitignore:
```diff
@@ -91,3 +91,6 @@ ENV/
 db.sqlite3
+
+#vscode
+.vscode
```
- Push a .env file template (with no confidential information) to github then add the .env file here since we don't want the confidential information we shall set in our machines for our project to be available to any one who has access to our github repo.
- Let the template have the name `.env-sample` then add some comments on the readme.MD file on how to update the  .env file when someone is trying to set up our project for the first time.

dmigwi reviewed (changes requested):
Fix the merge conflicts and the issues about security that I've raised.

Commit:
fix (database): set confidential db information as environment variables

- add .env file
- add env-sample file
- add instructions to setup the environment variables locally

[Fixes #159053979]

Commit:
fix merge conflicts
